### ISSUE TITLE
Conflict with react/jsx-one-expression-per-line

### ISSUE THREAD

**Issue by EdmundsEcho** [prettier/prettier#6456]
Thank you for this configuration.  It is really helpful.

**Prettier 1.18.2**
[Playground link](https://prettier.io/playground/#.....)
```sh
# Options (if any):
--single-quote
```

**Input:**
```jsx
          <Folder />
          {' '}          // <<< prettier/prettier wants me to delete the linebreak
          </TableCell>
```

**Output:**
```jsx
          <Folder />
{' '}
         </TableCell>

```
The linting message and fix go round and round :-(.  `react/jsx-one-expression-per-line` requires I put the second line, on a separate line.  `prettier/prettier` wants me to delete the linebreak between what is now the second line and the and the first line.

I believe that jsx rule makes sense.  

**Expected behavior:**
```jsx
         <Folder />
         {' '}          
         </TableCell>
```

Thank you for any guidance you can provide.

\- E

**Comment by lydell** [comment]
Hi! We recommend turning off all conflicting linting rules. See https://prettier.io/docs/en/integrating-with-linters.html

**Comment by EdmundsEcho** [comment]
Thank you for getting back to me so quickly.  I have those extensions up and running.  

... a sample of my `devDependencies`:
```
    "eslint-config-airbnb": "^18.0.1",
    "eslint-config-prettier": "^6.1.0",
    "eslint-config-react": "^1.1.7",
    "eslint-plugin-prettier": "^3.1.0",
    "prettier": "^1.18.2",
    "prettier-eslint": "^9.0.0",
```
my `plugins` property
```
    "plugins": [
      "react",
      "prettier"
    ],
```
and `extends` (with prettier having "the last word" :-) 
```
    "extends": [
      "airbnb",
      "eslint:recommended",
      "plugin:react/recommended",
      "plugin:prettier/recommended"
    ],
```
What do you think?

\- E

**Comment by lydell** [comment]
You need to add `"prettier/react"` to the `"extends"` array.

**Comment by EdmundsEcho** [comment]
Fantastic. Solved!  Thank you so much.  So just to confirm (teaching me to fish), I need to include both the npm `devDependencies` that describe the extra rules for `eslint` in addition to the plugins themselves. Yes?

Also, does the sequence in the array dictate any kind of source of truth priority (e.g., "last in")?

**Comment by lydell** [comment]
> I need to include both the npm `devDependencies` that describe the extra rules for `eslint` in addition to the plugins themselves. Yes?

I’m not sure what you mean?

> Also, does the sequence in the array dictate any kind of source of truth priority (e.g., "last in")?

https://eslint.org/docs/user-guide/configuring#extending-configuration-files

> The extends property value is either:

> - a string that specifies a configuration
> - an array of strings: each additional configuration extends the preceding configurations

**Comment by EdmundsEcho** [comment]
[extends] must describe (point to) references to rules.  This can include rules that only serve to supress potentially contradicting rules.  What was most confusing (I'm talking about in general for eslint configuation) was to figure out the relationship between [extends] [plugins] and the `devDependencies`.  Some of the dependencies imply that it is only a plugin, others just rules...

My learning and hoping you can confirm is this:

```
extends -> npm modules
["airbnb"] ->  [eslint-config-airbnb]
["eslint:recommended"] -> [eslint]
["plugin:react/recommended"] -> [eslint-plugin-react, eslint-plugin-react-hooks, eslint-config-react-app ...] (see react-scripts)
["plugin:prettier/recommended"] -> [eslint-plugin-prettier, prettier, prettier-eslint, eslint-config-prettier]
["prettier/react"] -> is included "for free" in ... ??

```
In other words, it seems like if we want to "extend" eslint, we have to load npm modules to define the rules.  It is not clear when we have to load more "plugins" and what rules the plugins come baked-in with.  The "configure" word does seem to indicate "coordinate" with other rules.  

Am I missing something in how the organization of the references generally work?

**Comment by lydell** [comment]
https://eslint.org/docs/user-guide/configuring#configuring-plugins

> ESLint supports the use of third-party plugins. Before using the plugin, you have to install it using npm.

To configure plugins inside of a configuration file, use the `plugins` key, which contains a list of plugin names. The `eslint-plugin-` prefix can be omitted from the plugin name.

https://eslint.org/docs/user-guide/configuring#using-a-shareable-configuration-package

> A sharable configuration is an npm package that exports a configuration object. Make sure the package has been installed to a directory where ESLint can require it.
> 
> The `extends` property value can omit the `eslint-config-` prefix of the package name.

https://eslint.org/docs/user-guide/configuring#using-the-configuration-from-a-plugin

> A plugin is an npm package that usually exports rules. Some plugins also export one or more named configurations. Make sure the package has been installed to a directory where ESLint can require it.
>
> The plugins property value can omit the `eslint-plugin-` prefix of the package name.
>
> The `extends` property value can consist of:
>
> - `plugin:`
> - the package name (from which you can omit the prefix, for example, `react`)
> - `/`
> - the configuration name (for example, `recommended`)
